activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb

```diff
@@ -96,12 +96,12 @@ def execute(sql, name = nil)
 
         def exec_query(sql, name = "SQL", binds = [], prepare: false)
           execute_and_clear(sql, name, binds, prepare: prepare) do |result|
-            types = {}
+            types = []
             fields = result.fields
             fields.each_with_index do |fname, i|
               ftype = result.ftype i
               fmod  = result.fmod i
-              types[fname] = get_oid_type(ftype, fmod, fname)
+              types << get_oid_type(ftype, fmod, fname)
             end
             build_result(columns: fields, rows: result.values, column_types: types)
```

Comment:
I think this `column_types:` parameter should be renamed to `types:`.

The default value is now of the wrong type too:

https://github.com/rails/rails/blob/a3f62cbbe76f7f75c8442d5bab13a8f1bdfd00a7/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L746